### PR TITLE
Fix bashism

### PR DESCRIPTION
**Pull request by bastien-roucaries**
Hi, 

Could you please check and merge this ?

For debian we will like to only use dash for this package

### CODE REVIEW

**Comment by bastien-roucaries** [comment]
Note that it is an incomplete fix bashism

**Comment by Schievel1** [comment]
Thanks for your contribution. 
Shouldn't the shebang then be /bin/sh as well?

https://github.com/Antynea/grub-btrfs/blob/465b56107f1a9a983e132d17441c2d63cdc16392/41_snapshots-btrfs#L1C1-L1C21

**Comment by Schievel1** [comment]
@bastien-roucaries could you explain that a bit more? You haven't fixed all bashisms in this, yet you did enough for dash to work with this. Does that mean dash works with some bashisms, not all? 

I still have this error when doing shellcheck -s sh:
```
In 41_snapshots-btrfs line 301:
        local path_snapshot=${snap[@]:13:${#snap[@]}}
        ^-----------------^ SC3043 (warning): In POSIX sh, 'local' is undefined.
                            ^-----------------------^ SC2068 (error): Double quote array expansions to avoid re-splitting elements.
                            ^-----------------------^ SC2124 (warning): Assigning an array to a string! Assign as array, or use * instead of @ to concatenate.
```

**Comment by bastien-roucaries** [comment]
@Schievel1 it is a first pass for eliminating bashism. For array I will need more review

**Comment by Schievel1** [comment]
Ok, I get it. Then you will follow up with more PRs regarding this? This is a ton of work to work around the arrays here, isn't it?}

**Comment by bastien-roucaries** [comment]
@Schievel1 not necesseraily if we could get the code clearer by refactoring

File: 41_snapshots-btrfs [hunk 1]
```diff
@@ -574,15 +574,15 @@ if [ -e "$grub_btrfs_directory/grub-btrfs.cfg" ]; then
...
+[ ! -d "$grub_btrfs_mount_point" ] && mkdir -p "$grub_btrfs_mount_point"
 mount -o ro,subvolid=5 /dev/disk/by-uuid/"$root_uuid" "$grub_btrfs_mount_point/" > /dev/null
 trap "unmount_grub_btrfs_mount_point" EXIT # unmounting mount point on EXIT signal
 count_warning_menuentries=0 # Count menuentries
 count_limit_snap=0 # Count snapshots
 check_uuid_required
 # Detects if /boot is a separate partition
-[[ "${GRUB_BTRFS_OVERRIDE_BOOT_PARTITION_DETECTION,,}" == "true" ]] && printf "Override boot partition detection : enable \n" >&2 && boot_separate;
-if [[ "$root_uuid" != "$boot_uuid" ]] || [[ "$root_uuid_subvolume" != "$boot_uuid_subvolume" ]]; then boot_separate ; else boot_bounded ; fi
+[ "$(echo "$GRUB_BTRFS_OVERRIDE_BOOT_PARTITION_DETECTION}" | tr '[:uper:]' '[:lower:]')" = "true" ] && printf "Override boot partition detection : enable \n" >&2 && boot_separate;
```
Comment (Schievel1) [inline 41_snapshots-btrfs:584]: this should be `[:upper:]` shouldn't it?

File: 41_snapshots-btrfs [hunk 2]
```diff
@@ -75,15 +75,15 @@ while getopts :V-: opt; do
 done
 
 ## Exit the script, if:
-[[ "${GRUB_BTRFS_DISABLE,,}" == "true" ]] && print_error "GRUB_BTRFS_DISABLE is set to true (default=false)"
+[ "$(echo "$GRUB_BTRFS_DISABLE" | tr '[:uper:]' '[:lower:]')" = 'true' ] && print_error "GRUB_BTRFS_DISABLE is set to true (default=false)"
```
Comment (Schievel1) [inline 41_snapshots-btrfs:78]: this should be [:upper:] shouldn't it?

Comment (bastien-roucaries) [reply 41_snapshots-btrfs:78]: yes typo

**Comment by Schievel1** [comment]
Thanks for your contribution. I am looking forward to test following PRs from you that purge bashisms :)

**Pull request merged**